### PR TITLE
fix(argos): waiting for images decoding

### PR DESCRIPTION
**Pull request by axel7083**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

### Test links

Deploy preview: https://deploy-preview-_____--docusaurus-2.netlify.app/

## Related issues/PRs

Fix https://github.com/facebook/docusaurus/issues/9805

### CODE REVIEW

**Comment by slorber** [comment]
Thanks

@gregberge wouldn't you want this to be included automatically in your Argos Playwright integration package?

https://github.com/argos-ci/argos-javascript/tree/main/packages/playwright

File: argos/tests/screenshot.spec.ts [hunk 1]
```diff
@@ -91,11 +91,21 @@ function waitForDocusaurusHydration() {
...
+    ),
+  );
+}
+
 function createPathnameTest(pathname: string) {
   test(`pathname ${pathname}`, async ({page}) => {
     const url = siteUrl + pathname;
     await page.goto(url);
     await page.waitForFunction(waitForDocusaurusHydration);
+    await waitForImageDecoding();
```
Comment (slorber) [inline argos/tests/screenshot.spec.ts:108]:
```suggestion
    await page.waitForFunction(waitForImageDecoding);
```

**Comment by slorber** [comment]
@axel7083 we'll probably add this on the Argos playwright integration, but in its current form, your code doesn't seem to fix the flakiness issue.

**Comment by axel7083** [comment]
> @axel7083 we'll probably add this on the Argos playwright integration, but in its current form, your code doesn't seem to fix the flakiness issue.

Yeah I thought it would, but apparently not - that's a bit annoying actually

**Comment by slorber** [comment]
Seems to be working, Argos only reports the expected visual diff in this PR now: https://github.com/facebook/docusaurus/pull/9868

**Pull request merged**